index.js

```diff
@@ -191,3 +204,11 @@ function zOrder(x, y) {
 
     return x | (y << 1);
 }
+
+function hasAllKeys(tags, requiredKeys) {
+    var tagKeys = tags.filter(function (_, i) { return i % 2 === 0; });
+    for (var i = 0, j = requiredKeys.length; i < j; i++) {
+        if (tagKeys.indexOf(requiredKeys[i]) === -1) return false;
+    }
+    return true;
+}
```

Comment:
With a hashmap (as suggested above), knowing that tag keys are unique in a feature, we can do it faster like this:

```js
var numKeys = 0;
for (var i = 0; i < tagKeys.length; i += 2) {
    if (requiredLookup[tagKeys[i]]) numKeys++;
}
return numKeys === numRequiredKeys;
```